### PR TITLE
Attic Air Sealing - Template Creating

### PR DESCRIPTION
**Pull request by sudhacheran**
New template for Attic Air Sealing Installation report

### CODE REVIEW

**Review by charliepnnl — CHANGES_REQUESTED** [review]

File: src/templates/doe_workflow_atticairsealing.mdx [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    <TextInput  label="Summary Comments" path="installation_summary_comments" />
+    
+  </Tab>
+ <Tab eventKey="report" style={{paddingTop:"1rem"}} title="Report">
+    <PrintSection label="Print Report">
+    ---
+    # Attic Air Sealing Upgrade Photo Report
+
+    **Installer**:<br/>
+    {props.data.installer?.name ? <span>&nbsp;&nbsp;&nbsp;&nbsp;{props.data.installer?.name}</span> : null}<br/>
```
[the remark this answers is not in the document]
Comment (sudhacheran) [reply src/templates/doe_workflow_atticairsealing.mdx:154]: Moving the <br/> after </span> didn't work. Error thrown as 
_Module build failed (from ./node_modules/@mdx-js/loader/index.cjs):
doe_workflow_hpwh.mdx:225:100: Could not parse expression with acorn: Adjacent JSX elements must be wrapped in an enclosing tag_

Also, placing with <spa><br/></span> is not rendering it properly. It adds a space after "Installer:" .

So I have placed with additional check. I see it a redundant check, please let me know your suggestion .

File: src/templates/doe_workflow_atticairsealing.mdx
[the remark this answers is not in the document]
Comment (sudhacheran) [reply src/templates/doe_workflow_atticairsealing.mdx:1]: Done

File: src/templates/templates_config.ts [hunk 2]
```diff
@@ -15,6 +18,14 @@ const templatesConfig : TemplateConfig = {
   doe_workflow_hpwh: {
     title: "Heat Pump Water Heater",
     template: DOEWorkflowHPWHTemplate,
+  },
+  doe_workflow_attic_sealing: {
```
[the remark this answers is not in the document]
Comment (sudhacheran) [reply src/templates/templates_config.ts]: done

File: src/templates/doe_workflow_atticairsealing.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (charliepnnl) [reply src/templates/doe_workflow_atticairsealing.mdx]: Ok

File: src/templates/doe_workflow_atticairsealing.mdx [hunk 3]
```diff
@@ -0,0 +1,314 @@
+<Tabs>
+<Tab eventKey="project" style={{paddingTop:"1rem"}} title="Project">
```
[the remark this answers is not in the document]
Comment (sudhacheran) [reply src/templates/doe_workflow_atticairsealing.mdx]: done

File: src/templates/doe_workflow_atticairsealing.mdx [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+    The data for this tab should be know prior to the initial site visit.
+
+    *The Installer inputs below are all optional, although you will want to 
+    fill in at least one for the report.*
+
+    ##### Installer
```
[the remark this answers is not in the document]
Comment (sudhacheran) [reply src/templates/doe_workflow_atticairsealing.mdx]: done

File: src/templates/doe_workflow_atticairsealing.mdx [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+  
+    ##### Attic Floor Prepared for Air Sealing
+    <PhotoInput id="attic_floor_prepared" label=" Attic Floor Preparation – Photo">
+      Photo of attic floor prepared for attic air sealing such as cleaned, insulation removed
+    </PhotoInput>
+  
+  </Tab>
+  <Tab eventKey="installation" style={{paddingTop:"1rem"}} title="Installation 1">
+
+   <DateInput label="Installation Date" path="install_date" />
```
[the remark this answers is not in the document]
Comment (sudhacheran) [reply src/templates/doe_workflow_atticairsealing.mdx]: done

File: src/templates/doe_workflow_atticinsulation.mdx
[the remark this answers is not in the document]
Comment (sudhacheran) [reply src/templates/doe_workflow_atticinsulation.mdx:1]: done

File: src/templates/doe_workflow_atticairsealing.mdx
[the remark this answers is not in the document]
Comment (sudhacheran) [reply src/templates/doe_workflow_atticairsealing.mdx:1]: done

File: src/templates/doe_workflow_attic_air_sealing.mdx [hunk 6]
```diff
@@ -0,0 +1,291 @@
+<Tabs>
+<Tab eventKey="project" title="Project">
+    The data for this tab should be know prior to the initial site visit.
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_air_sealing.mdx:3]: Missing "n": "should be _known_"

File: src/templates/doe_workflow_attic_air_sealing.mdx [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+  <Tab eventKey="assessment" title="Assessment">
+    The data for this tab should be collected during the initial site visit.
+     
+    ## Building 
+    <PhotoInput id="building_number_photo" label="Building Number – Photo">
+      Provide a photo of the building that shows the street number.
+    </PhotoInput>
+
+    ## Blower Door
+    <PhotoInput id="blower_door_setup" label="Blower Door Setup – Photo">
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_air_sealing.mdx:16]: All photo ids should end with "_photo" to match our earlier system.

File: src/templates/doe_workflow_attic_insulation.mdx [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+        </PhotoInput>
+
+        Skip photo if feature is N/A to house.
+        
+        <PhotoInput id="insulation_measuring_rulers " label="Insulation Measuring Rulers  – Photo ">
+            Overview of insulation measuring rulers, one installed every 300 sq. ft. visible 
+            from attic entrance (photo taken from attic entrance).     
+        </PhotoInput>
+
+        <PhotoInput id="rvalue" label="R-value Per Inch  – Photo ">
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_insulation.mdx:24]: Let's make the id for this more descriptive. Maybe "rvalue_table_photo".

File: src/templates/doe_workflow_attic_insulation.mdx [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+
+        <PhotoInput id="storage_platform" label="Storage Platform – Photo ">
+        Photo of storage platform insulation detail    
+        </PhotoInput>
+
+        <PhotoInput id="wide_angle_photo" label="Wide Angle – Photo ">
+        Wide angle photo showing insulation uniformity with rulers visible
+        </PhotoInput>
+
+         <PhotoInput id="atticdoor_hatch" label="Attic Door/Hatch - Photo ">
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_insulation.mdx:48]: The id should be "attic_door_hatch_photo".

File: src/templates/doe_workflow_attic_insulation.mdx [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+        Photo of insulation on attic door/hatch
+        </PhotoInput>
+    
+    </Tab>
+    <Tab eventKey="postassessment" title="Post Assessment">
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_insulation.mdx:53]: Just for consistency, the event key should probably be "post_assessment".

**Review by charliepnnl — CHANGES_REQUESTED** [review]
Just a few changes requested

File: src/templates/doe_workflow_attic_air_sealing.mdx (hunk 6 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_air_sealing.mdx]: changed in three templates

File: src/templates/doe_workflow_attic_insulation.mdx (hunk 8 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_insulation.mdx]: done

File: src/templates/doe_workflow_attic_air_sealing.mdx (hunk 7 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_air_sealing.mdx]: Update the id of all PhotoInput and Photo components. Tested the changes proper update in the database.

File: src/templates/doe_workflow_attic_insulation.mdx (hunk 9 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_insulation.mdx]: Done

File: src/templates/doe_workflow_attic_insulation.mdx (hunk 10 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_insulation.mdx]: done

File: src/templates/doe_workflow_attic_air_sealing.mdx [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+    ## Blower Door
+    <PhotoInput id="blower_door_setup_photo" label="Blower Door Setup – Photo">
+        Photo of the blower door setup before the air sealing insulation work
+    </PhotoInput>
+    <PhotoInput id="cfm50_photo" label="CFM50 – Photo">
+        Photo of the manometer showing CFM50 of air leakage before air sealing and insulation work was performed
+    </PhotoInput>
+    | ACH50 Calculator                                                                                                                                                                                                                                           |
+    | ----------------------------
+    | <NumberInput label="CFM50" min={0} path="ACH50_calculator.cfm50"  />
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_air_sealing.mdx:24]: For consistency, let's only use lower-case in paths, so "ach50_calculator.cfm50" here. Similarly below.

File: src/templates/doe_workflow_attic_air_sealing.mdx [hunk 12]
```diff
@@ -0,0 +1,291 @@
...
+    <Photo id="blower_door_setup_photo" label="Blower Door  – Photo" required>
+        Photo of the blower door setup before the air sealing insulation work
+    </Photo>
+    <Photo id="cfm50_photo" label="CFM50 – Photo" required>
+        Photo of the manometer showing CFM50 of air leakage before air sealing and insulation work was performed
+    </Photo>
+
+    | ACH50 Calculation                                                                                                                                                                                                                                           |
+    | ----------------------------
+    | CFM50 = { props.data.ACH50_calculator?.cfm50 ? props.data.ACH50_calculator?.cfm50 : null }
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_air_sealing.mdx:180]: Lower-case ach50 here too.

File: src/templates/doe_workflow_attic_air_sealing.mdx [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+    | Building Volume (ft&#179;) = { props.data.ACH50_calculator?.volume ? props.data.ACH50_calculator?.volume : null }
+    | ACH50 = { (props.data.ACH50_calculator?.cfm50 && props.data.ACH50_calculator?.volume) ? ((60 * props.data.ACH50_calculator?.cfm50) / props.data.ACH50_calculator?.volume).toFixed(2) : null }
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_air_sealing.mdx:182]: I'm glad you thought to use `toFixed(2)`!

File: src/templates/doe_workflow_attic_insulation.mdx [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+            Photo of Blower door setup 
+        </PhotoInput>
+    
+        <PhotoInput id="blowerdoor_cfm50_photo" label="Manometer CFM50 - Photo">
+            Photo of blower door manometer showing CFM at 50 Pa read out
+        </PhotoInput>
+
+        | Leakage Reduction Calculator                                                                                                                                                                                                                                           |
+        | ----------------------------
+        | <NumberInput label="Final CFM50" min={0} path="ACH50_calculator.final_cfm50"  />
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_insulation.mdx:65]: Use lower-case "ach50" here too.

File: src/templates/doe_workflow_attic_insulation.mdx [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+            Photo of blower door setup
+        </Photo>
+    
+        <Photo id="blowerdoor_cfm50_photo" label="Manometer CFM50 - Photo" required>
+            Photo of blower door manometer showing CFM at 50 Pa read out
+        </Photo>
+
+        | Leakage Reduction Calculator                                                                                                                                                                                                                                           |
+        | ----------------------------
+        | Final CFM50 = { props.data.ACH50_calculator?.final_cfm50 ? props.data.ACH50_calculator?.final_cfm50 : null }
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_insulation.mdx:165]: And lower-case "ach50" here too.

File: src/templates/doe_workflow_attic_insulation.mdx [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+        <Photo id="blowerdoor_cfm50_photo" label="Manometer CFM50 - Photo" required>
+            Photo of blower door manometer showing CFM at 50 Pa read out
+        </Photo>
+
+        | Leakage Reduction Calculator                                                                                                                                                                                                                                           |
+        | ----------------------------
+        | Final CFM50 = { props.data.ACH50_calculator?.final_cfm50 ? props.data.ACH50_calculator?.final_cfm50 : null } 
+        | Initial CFM50 = { props.data.ACH50_calculator?.ini_cfm50 ? props.data.ACH50_calculator?.ini_cfm50 : null }  
+        | Building Volume (ft&#179;) = { props.data.ACH50_calculator?.volume ? props.data.ACH50_calculator?.volume : null } 
+        | Initial ACH50 = { props.data.ACH50_calculator?.ini_cfm50 && props.data.ACH50_calculator?.volume ? ((60 * props.data.ACH50_calculator.ini_cfm50) / props.data.ACH50_calculator.volume).toFixed(2) : null }
```
Comment (charliepnnl) [inline src/templates/doe_workflow_attic_insulation.mdx:168]: let's write out `.initial_cfm50` it is only three more characters.

**Review by charliepnnl — APPROVED** [review]
Approved with just a few comments that need to be addressed.
I focused on naming of all the paths and attachment ids this time.

Please merge into main after making the requested changes.

File: src/templates/doe_workflow_attic_air_sealing.mdx (hunk 11 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_air_sealing.mdx]: changed to lowercase in both the new templates

File: src/templates/doe_workflow_attic_air_sealing.mdx (hunk 12 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_air_sealing.mdx]: done

File: src/templates/doe_workflow_attic_insulation.mdx (hunk 14 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_insulation.mdx]: done

File: src/templates/doe_workflow_attic_insulation.mdx (hunk 15 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_insulation.mdx]: done

File: src/templates/doe_workflow_attic_insulation.mdx (hunk 16 above)
Comment (sudhacheran) [reply src/templates/doe_workflow_attic_insulation.mdx]: done

**Review by charliepnnl — APPROVED** [review]

**Pull request merged**